os/arch/arm/src/imxrt/imxrt_serial.c

```diff
@@ -1520,12 +1528,11 @@ int up_putc(int ch)
 
 	if (ch == '\n') {
 		/* Add CR */
-
-		up_lowputc('\r');
+		imxrt_lowputc('\r');
 	}
 
-	up_lowputc(ch);
-	imxrt_restoreuartint(priv, intset);
+	imxrt_lowputc(ch);
```

Comment:
Here is not in CONFIG_DEBUG conditional. But at above, definition of this function is in it. When debug is disabled, is it ok?

Reply:
We have this code in imxrt_lowputc.h. So it will be ok if CONFIG_DEBUG is disabled.

`#if defined(HAVE_LPUART_DEVICE) && defined(CONFIG_DEBUG)
'void imxrt_lowputc(int ch);
'#else
'#define imxrt_lowputc(ch)
'#endif
`

Reply:
We have this code in imxrt_lowputc.h. So it will be ok if CONFIG_DEBUG is disabled.

#if defined(HAVE_LPUART_DEVICE) && defined(CONFIG_DEBUG)
void imxrt_lowputc(int ch);
#else
#define imxrt_lowputc(ch)
#endif

Reply:
In compilation point of view, it's ok. But is there any reason to block it with debug disabled?

Reply:
I dont think it needs to depend on CONFIG_DEBUG. I will remove the condition for CONFIG_DEBUG and keep the check for HAVE_LPUART_DEVICE.